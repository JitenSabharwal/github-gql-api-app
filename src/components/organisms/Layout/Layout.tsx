import React from "react"
import styled from "styled-components"
import Header from "../../molecules/Header"
const Root = styled.div``
const Body = styled.div`
  margin: 24px;
`
export default function Layout({
  children,
  onBackClick,
}: {
  children: React.ReactNode
  onBackClick?: () => void
}) {
  return (
    <Root>
      <Header onBackClick={onBackClick} heading={"GitHub Issue Search"} />
      <Body>{children}</Body>
    </Root>
  )
}
