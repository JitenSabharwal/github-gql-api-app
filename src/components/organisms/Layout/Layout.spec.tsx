import Layout from "./Layout"
import React from "react"
import { shallow } from "enzyme"
describe("Layout", () => {
  let container: React.ReactElement
  beforeAll(
    () =>
      (container = (
        <Layout onBackClick={() => {}}>
          <div />
        </Layout>
      ))
  )
  it("Should render my component", () => {
    const component = shallow(container)
  })
  it("Should render initial layout", () => {
    const component = shallow(container)
    expect(component.getElements()).toMatchSnapshot()
  })
  it("Should render initial layout without back button", () => {
    const component = shallow(
      <Layout>
        <div />
      </Layout>
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
