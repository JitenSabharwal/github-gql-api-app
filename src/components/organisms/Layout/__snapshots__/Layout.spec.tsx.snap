// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should render initial layout 1`] = `
Array [
  <styled.div>
    <Header
      heading="GitHub Issue Search"
      onBackClick={[Function]}
    />
    <styled.div>
      <div />
    </styled.div>
  </styled.div>,
]
`;

exports[`Layout Should render initial layout without back button 1`] = `
Array [
  <styled.div>
    <Header
      heading="GitHub Issue Search"
    />
    <styled.div>
      <div />
    </styled.div>
  </styled.div>,
]
`;
